src/ml_board/backend/websocket_api/websocket_server.py

```diff
@@ -59,100 +143,26 @@ def _send_event_history_to_client(self, client_id: str, room_id: str):
-                    )
-                join_room(room)
-            print(f"Client {client_id} joined rooms: {rooms()}")
-            print(rooms_to_join)
-            self.emit_server_log_message(f"Client {client_id} joined rooms: {rooms()}")
-            for room in rooms_to_join:
-                self._send_event_history_to_client(client_sid, room)
-
-        @self._socketio.on("leave")
-        def on_leave():
-            self._client_sids.remove(request.sid)
-            # TODO  leave all rooms
-            # leave_room(message['room'])
-            self.emit_server_log_message("You are now disconnected.")
-            disconnect()
-
-        @self._socketio.on("mlgym_event")
-        def on_mlgym_event(data):
-            grid_search_id = data["payload"]["grid_search_id"]
-            if data["event_type"] in set(["experiment_status", "job_status", "experiment_config", "evaluation_result"]):
-                print("mlgym_event: " + str(data))
-                if grid_search_id not in self._room_id_to_event_storage:
-                    self._room_id_to_event_storage[grid_search_id] = EventStorageFactory.get_disc_event_storage(
-                        parent_dir=self._top_level_logging_path, event_storage_id=grid_search_id
-                    )
-                event_id = self._room_id_to_event_storage[grid_search_id].add_event(data)
-                emit('mlgym_event', {'event_id': event_id, 'data': data}, to=grid_search_id)
-            else:
-                print(f"Unsupported event_type {data['event_type']}")
-
-        # @socketio.event
-        # def disconnect_request():
-        #     @copy_current_request_context
-        #     def can_disconnect():
-        #         disconnect()
-
-        #     session['receive_count'] = session.get('receive_count', 0) + 1
-        #     # for this emit we use a callback function
-        #     # when the callback function is invoked we know that the message has been
-        #     # received and it is safe to disconnect
-        #     emit('my_response',
-        #          {'data': 'Disconnected!', 'count': session['receive_count']},
-        #          callback=can_disconnect)-
-
-        @self._socketio.on("ping")
-        def on_ping():
-            emit("pong")
-
-        @self._socketio.on("client_connected")
-        def on_client_connected():
-            print(f"Client with SID {request.sid} connnected.")
-            self.emit_server_log_message(f"Client with SID {request.sid} connnected.")
-
-        @self._socketio.on("client_disconnected")
-        def on_client_disconnected():
-            print("Client disconnected", request.sid)
-            self._client_sids.remove(request.sid)
+####################################################################################################
+# Flask HTTP Server
+def create_flask_server() -> Flask:
+    # create Flask server
+    _app: Flask = Flask(__name__, template_folder="template")
+    _app.secret_key = 'secret!'
+    return _app
```

Comment:
where was this previously implemented?

Reply:
https://flask.palletsprojects.com/en/2.3.x/config/

Reply:
No where I took it from my implementation of the [dummy_server](https://github.com/mlgym/mlgym/blob/e3455fcfac62ed3477e6592bef4969908bc0ca19/src/ml_board/backend/dummy_server/dummy_server.py#LL14C1-L14C1)

based on [this](https://www.reddit.com/r/flask/comments/m0z7s1/need_some_help_understanding_the_use_of_a_flask/) and [this](https://stackoverflow.com/questions/34902378/where-do-i-get-secret-key-for-flask), I see setting the secret to a random string each time is not a bad idea for now, what do you think?

 `_app.secret_key = os.urandom(24)`

Reply:
ok that sounds like a reasonable choice for now :)